frameworks/elastic/src/main/dist/elasticsearch.yml

```diff
@@ -21,3 +21,7 @@ node.data: {{DATA_ENABLED}}
 node.ingest: {{INGEST_ENABLED}}
 metrics.statsd.host: {{STATSD_UDP_HOST}}
 metrics.statsd.port: {{STATSD_UDP_PORT}}
+gateway.expected_master_nodes: 3
```

Comment:
It will, yes. According to Elasticsearch docs, a full cluster restart is required when these settings change. https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-gateway.html
I will ask around though to see if that really has to happen on every single node type. Seems like the sort of thing that could be specified just on the masters' `elasticsearch.yml` files.